[quant][refactor] Remove register api and rename get_*_mapping to get_default_*_mapping

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #46355 [quant][eagermode] Add additional_fuser_method_mapping to config
* #46346 [quant][graphmode][fx] Add support for additional_{fusion/quant}_pattern
* #46345 [quant][graphmode][fx] Add support for additional_fuse_method_mapping
* #46344 [quant][graphmode][fx] Add additional_qat_module_mapping
* #46338 [quant][graphmode][fx] Add additional_object_mapping argument to convert
* **#46337 [quant][refactor] Remove register api and rename get_*_mapping to get_default_*_mapping**
* #46298 [quant][graphmode][fx] Support non_traceable_module/module_class

Summary:
We plan to pass around the mappings instead of using global registration api to keep
the mappings local to the transformations user is performing

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317436](https://our.internmc.facebook.com/intern/diff/D24317436)

jerryzh168 committed:
[quant][refactor] Remove register api and rename get_*_mapping to get_default_*_mapping

Summary:
We plan to pass around the mappings instead of using global registration api to keep
the mappings local to the transformations user is performing

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][refactor] Remove register api and rename get_*_mapping to get_default_*_mapping"


Summary:
We plan to pass around the mappings instead of using global registration api to keep
the mappings local to the transformations user is performing

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317436](https://our.internmc.facebook.com/intern/diff/D24317436)

[ghstack-poisoned]

vkuzo reviewed (approved):
lg, would be good to double check there are no fb-only deps too

jerryzh168 committed:
Update on "[quant][refactor] Remove register api and rename get_*_mapping to get_default_*_mapping"


Summary:
We plan to pass around the mappings instead of using global registration api to keep
the mappings local to the transformations user is performing

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24317436](https://our.internmc.facebook.com/intern/diff/D24317436)

[ghstack-poisoned]
